### PR TITLE
fix: [CL-008] remove duplicate wrapping

### PR DESCRIPTION
**Pull request by howydev**
to reviewers: best to review other deferred action changes first since that can impact this

### CODE REVIEW

File: src/account/SemiModularAccountBase.sol [hunk 1]
```diff
@@ -112,7 +112,13 @@ abstract contract SemiModularAccountBase is ModularAccountBase {
         if (sigValidation.eq(FALLBACK_VALIDATION)) {
             address fallbackSigner = _getFallbackSigner();
 
-            if (_checkSignature(fallbackSigner, _replaySafeHash(hash), signature)) {
+            // If called during validateUserOp, this implies that we're doing a deferred validation installation.
+            // In this case, as the hash is already replay-safe, we don't need to wrap it.
+            if (msg.sig != this.validateUserOp.selector) {
```
Comment (adamegyed) [inline src/account/SemiModularAccountBase.sol:117]: I'm not sure this is the right way to do this switching logic, it seems ok on first glance but feels a bit fragile. Could we make the caller of `_exec1271Validation` perform the wrapping in the cases where it is needed, and skip it in the deferred action case?

Comment (Zer0dot) [reply src/account/SemiModularAccountBase.sol:117]: Could you elaborate on this point? 

Would you instead not wrap it in `validateUserOp()`'s deferred validation section, but wrap in `isValidSignature()`?

Comment (howydev) [reply src/account/SemiModularAccountBase.sol:117]: Those approaches require modifying  `_isValidSignature` or `_validateUserOp` in ModularAccountBase (function doesn't exist in SMA) which would be a huge change

Agree that it feels jank, but this is the least complex solution. Happy to change my mind if there's an easier solution to the above or if specific issues about this way of doing things are found

Comment (adamegyed) [reply src/account/SemiModularAccountBase.sol:117]: Good points, yes that makes sense. I'm ok with this solution.

**Review by adamegyed — APPROVED** [review]

**Pull request merged**